bot/exts/filtering/_filter_lists/extension.py

```diff
@@ -116,3 +127,80 @@ async def actions_for(
         ctx.blocked_exts |= set(not_allowed)
         actions = self[ListType.ALLOW].defaults.actions if ctx.event != Event.SNEKBOX else None
         return actions, [f"`{ext}`" if ext else "`No Extension`" for ext in not_allowed], {ListType.ALLOW: triggered}
+
+
+class EmbedFileHandler(commands.Cog):
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    async def _convert_attachment(attachment: discord.Attachment) -> paste_service.PasteFile:
+        encoding = re.search(r"charset=(\S+)", attachment.content_type).group(1)
+        file_content = (await attachment.read()).decode(encoding)
+        return paste_service.PasteFile(content=file_content, name=attachment.filename)
+
+    @commands.Cog.listener()
+    async def on_message(self, message: discord.Message) -> None:
+        # Check if the message contains an embedded file and is not sent by a bot
+        if message.author.bot or not message.attachments:
+            return
+
+        bot_reply = await message.reply(f"React with {PASTEBIN_UPLOAD_EMOJI} to upload your file to our paste bin")
+        await bot_reply.add_reaction(PASTEBIN_UPLOAD_EMOJI)
+
+        def wait_for_upload_permission(reaction: discord.Reaction, user: discord.User) -> bool:
+            return (
+                reaction.message.id == bot_reply.id
+                and str(reaction.emoji) == PASTEBIN_UPLOAD_EMOJI
+                and user == message.author
+            )
+
+        try:
+            # Wait for the reaction with a timeout of 60 seconds
+            await self.bot.wait_for("reaction_add", timeout=60.0, check=wait_for_upload_permission)
+        except TimeoutError:
+            await bot_reply.edit(content=f"~~{bot_reply.content}~~")
+            await bot_reply.clear_reactions()
+            return
+
+        logging.info({f.filename: f.content_type for f in message.attachments})
+
+        files = [
+            await self._convert_attachment(f)
+            for f in message.attachments
+            if f.content_type.startswith("text")
+        ]
+
+        try:
+            async with aiohttp.ClientSession() as session:
+                paste_response = await paste_service.send_to_paste_service(files=files, http_session=session)
+        except (paste_service.PasteTooLongError, ValueError):
+            # paste is too long
```

Comment:
this comment is a bit WET 🍋 

it's not a surprise that `PasteTooLongError` means the paste is too long 😁

Reply:
I think that comment is an artefact of when I found out that a long paste can also raise ValueError.